src/sleep/sleep.c

```diff
@@ -224,20 +224,23 @@ static int rtc_write_wake_alarm(uint64_t sec) {
         return 0;
 }
 
-static int execute_s2h(usec_t hibernate_delay_sec) {
-
-        _cleanup_strv_free_ char **hibernate_modes = NULL, **hibernate_states = NULL,
-                                 **suspend_modes = NULL, **suspend_states = NULL;
+static int execute_s2h(SleepConfig *sleep_config) {
+        char **suspend_modes       = NULL,
+             **suspend_states      = NULL,
+             **hibernate_modes     = NULL,
+             **hibernate_states    = NULL;
+        usec_t hibernate_delay_sec = 0;
         usec_t original_time, wake_time, cmp_time;
         int r;
 
-        r = parse_sleep_config("suspend", NULL, &suspend_modes, &suspend_states, NULL);
-        if (r < 0)
-                return r;
+        if (!sleep_config)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "No sleep config provided!");
```

Comment:
This is an internal API (even static!), so just use `assert(sleep_config)` here.